### ISSUE TITLE
PHPMailer::ENCRYPTION_STARTTLS

### ISSUE THREAD

**Issue by ueen** [PHPMailer/PHPMailer#2002]
Updated to the latest version via composer and set everything up according to the documentation, this line always provoked a 504 gateway timeout
`$mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS; `

so i changed it to
`$mail->SMTPSecure = 'ssl';`

and suddently it works, whats that about?

**Comment by Synchro** [comment]
It just means that _your_ mail server supports SMTPS and doesn't support SMTP+STARTTLS for encryption; it's nothing to do with PHPMailer. The correct constant to use is:

    $mail->SMTPSecure = PHPMailer::ENCRYPTION_SMTPS;

**Comment by ueen** [comment]
I do not think, that this is correct, my mail server DOES use SSL/TLS on Port 465
And why would you prefer STARTTLS over SSL?

**Comment by Synchro** [comment]
SMTPS and SMTP+STARTTLS are **both** encryption mechanisms for SMTP, and both use TLS they are essentially equally secure. As I said, SMTPS on port 465 was deprecated in 1998, because it was replaced by SMTP+STARTTLS, however, it is still in common use (mainly because of Microsoft continuing to use it on Exchange), it has also been mostly resurrected by [RFC8314](https://tools.ietf.org/html/rfc8314).

PHPMailer use SMTP+STARTTLS as the default, because that's what the standards have said to use for the last 20 years. If _your_ mail server does support that, it's fine, but you should amend *your* code to deal with the difference, not PHPMailer itself.

**Comment by ueen** [comment]
Thanks for your input, I learned quite a bit researching this, but as you yourself said, in respect to RFC8314, nowadays the port 465 is used for implicit SSL/TLS which is superior to STARTTLS, because it doesn't require cleartext communication, as clearly laid out in the very document you linked!
```
Connections to Mail Submission Servers and Mail Access Servers be made using "Implicit TLS" (as defined below), in preference to connecting to the "cleartext" port and negotiating TLS using the STARTTLS command or a similar command.
```

**Comment by ueen** [comment]
But what why is it unadvisable to use the latest and most secure option: implicit TLS on port 465 as unambiguously stated in RFC8314?

**Comment by Synchro** [comment]
Just because a server supports old SMTPS does not mean that it supports RFC8314 submission via implicit TLS on port 465 because they are not the same thing. Just because a server accepts connections on port 465 doesn't mean it supports submission.

This is one of the problems with changing SMTP - there are many things that could be achieved if you could persuade the entire internet to upgrade their mail servers. Over here, in reality, we want things to work with minimal hassle and maximum reliability. You are of course free to change it in your code, but I'm not changing the default for now.

**Comment by ueen** [comment]
I think you confuse smtps with SMTPS(=RFC8314) (see https://en.wikipedia.org/wiki/SMTPS)
And as smtps was deprecated 1998, I would assume that Mail-Servers that were updated since then actually support SMTPS and not smtps, and that does seem to be the case, as all Mail-Servers I checked do support it, including my shared server, my personal e-mail, google-mail.
I do understand that there are still obsolete Mail-Servers that use STARTTLS, but as this is discouraged I don't really see why PHPMail should perpetuate that, rather than encouraging SMTPS(=RFC8314) as the default.
I think it should at least be mentioned in the documentation, that STARTTLS on port 587 is in fact deprecated and implicit TLS on port 465 should be used whenever possible :)

**Comment by Synchro** [comment]
No, I really don't confuse them. You seem to be under the illusion that everybody in the world upgraded their mail servers within the last 2 years.

I don't know how many times I need to say this but **SMTP over implicit TLS on port 465 is not the same thing as RFC8314**, which is also not the same things as SMTPS. You are misinterpreting that description - notice that there is no RFC for SMTPS because it is exactly SMTP over TLS

It's not PHPMailer's place to try to push this standard; that's something that **only** servers can do. If PHPMailer was an SMTP server, I would absolutely agree with you – **but it's not** – it's a client that needs to prioritise reliability and interoperability over other concerns.

**Comment by ueen** [comment]
Yea all Mail-Servers I worked with encouraged the use of implicit TLS on port 465 according to RFC8314, so I was confused by the obsolete STARTTLS, obviously, thats only my little experience. I still think its reasonable to assume, that a lot of major Mail-Servers are up to date and comply with RFC8314, so I think its worth to mention in the documentation, for all the newcomers, who are not used to bother with Mail-Servers from 1998.
Thanks for taking the time to explain though, have a nice evening :)

**Comment by Synchro** [comment]
RFC8314 has existed for 2 years. Prior to that there had not been a standardised way *at all* to use SMTPS on port 465 for 20 years. Some mail servers did however arbitrarily implement what amounts to RFC6409 on that port. RFC8314 standardised that use. That's all it did. When you consider that a fair proportion of mail servers still don't support TLS *at all*, making an assumption that they will *all* support something relatively new is simply a recipe for code that's broken by default. Remember that GoDaddy, the world's biggest shared hosting service, doesn't use either encryption or authentication for SMTP (and breaks if you try), and making the change you propose would break PHPMailer for every one of the millions of users on there. You can't drive server adoption rates from the client side; you can however _accommodate_ server changes, which is what PHPMailer does.

Your case is also very unusual – not supporting RFC6409 means that there is no standard basis on which your mail server can support SMTP encryption using mail software more than 2 years old.

**Comment by ueen** [comment]
I don't claim to have full inside into this, I just wanted to offer my perspective, I do understand your concern.

My server also supports unencrypted communication, but I wouldn't dare to employ that.

**Comment by still-dreaming-1** [comment]
Given everything discussed here, is the gmail.phps example still the most recommended way to send mail through Gmail's SMTP service when not using OAUTH? It uses PHPMailer::ENCRYPTION_STARTTLS with port 587. This seems relevant to this dicussion because the open issue of what the capabilities of the mail server are is less open since we are explicitly taling about their service. Would it be more secure to set `$mail->SMTPSecure = ENCRYPTION_SMTPS::ENCRYPTION_SMTPS` and use port 465 as the code comment in the primary README example now encourages? I'm not making a claim, just asking a question, as I don't really understand this topic very well.

**Comment by ueen** [comment]
Simple Explanation: 
STARTTLS starts communicating with the mailserver unencrypted and is then negotiating an encrypted connecting
SSL/TLS SMTP (according to latest RFC8314): all communication is encrypted
So obviously the later is to favour if available, if not STARTTLS is fine, but obsolete.

**Comment by ueen** [comment]
Specifically Google provides both options
https://stackoverflow.com/questions/15146530/do-yahoo-gmail-smtp-server-uses-implicit-or-explicit-ssl (STARTTSL = explicit), as explored earlier, implicit SSL SMTP via port 465 is more secure.

**Comment by Synchro** [comment]
From a security point of view, they're identical, and the difference is academic. **Both** expose the host name and that's all (so long as you do actually issue a STARTTLS!), but RFC8314 is much harder to debug problems with, which is a _very_ big issue as far as PHPMailer is concerned. The difference is really not worth worrying about.
